src/irmin/tree.ml

```diff
@@ -1483,7 +1488,9 @@ module Make (P : Private.S) = struct
       | `Node n ->
           let* m = Node.to_map n in
           let bindings = m |> get_ok |> StepMap.bindings in
-          (node [@tailcall]) [] bindings (fun n -> k (`Tree n))
+          (node [@tailcall]) [] bindings (fun n ->
+              let n = List.sort (fun (s, _) (s', _) -> compare_step s s') n in
```

Comment:
Yes: we rely on this ordering implicitly in tests, since equality / pretty-printing is derived automatically for concrete trees from the type. We could change the generic operations to sort the bindings first, but it's simplest just to have a well-defined ordering when going from regular trees.